fix: inject correct MyInfo field titles in react

justynoh opened the pull request:
## Problem
Some MyInfo fields had copy edits to be made. This PR edits them.

Closes #4628 

EDIT (19 Oct 2022): Found a bug, there's backward incompatibility because the react app injects different myinfo field titles than angular does into the form definition. This PR patches this. Releasing at `v6.7.2`.


## Solution
Modified the edit drawer header to be identical to the field list drawer. Also updated some grammatical mistakes.

**Breaking Changes** 
- No - this PR is backwards compatible

justynoh committed:
chore: copy changes to MyInfo fields

karrui commented on shared/constants/field/myinfo/index.ts:
```diff
@@ -49,7 +49,7 @@ export const types: MyInfoFieldBlock[] = [
   {
     name: MyInfoAttribute.DateOfBirth,
-    value: 'Date of birth',
+    value: 'Date of Birth',
```
This is a shared file. Will this affect anything on the server? Is this truly backwards compatible?

justynoh replied:
yes and no :sob_smile:

justynoh commented:
not a blocker for react, converting to draft first

dependabot[bot] committed:
chore(deps-dev): bump eslint-plugin-jest from 27.1.1 to 27.1.3 (#5206)

Bumps [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) from 27.1.1 to 27.1.3.
- [Release notes](https://github.com/jest-community/eslint-plugin-jest/releases)
- [Changelog](https://github.com/jest-community/eslint-plugin-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jest-community/eslint-plugin-jest/compare/v27.1.1...v27.1.3)

---
updated-dependencies:
- dependency-name: eslint-plugin-jest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

justynoh commented:
rebasing

justynoh committed:
chore: revert values for myinfo field types

justynoh committed:
chore: revert values for myinfo field types again

justynoh committed:
fix: switch reference for MyInfo edit drawer titles and field titles

justynoh committed:
chore: bump version to v6.7.2

justynoh commented:
sorry the jest bump got muddled in the rebase :(
